cmd/jaeger/internal/extension/jaegerstorage/config.go

```diff
@@ -4,119 +4,47 @@
 package jaegerstorage
 
 import (
-	"errors"
-	"fmt"
-	"reflect"
-	"time"
-
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/confmap/xconfmap"
 
+	"github.com/jaegertracing/jaeger/cmd/internal/storageconfig"
 	"github.com/jaegertracing/jaeger/internal/config/promcfg"
-	cascfg "github.com/jaegertracing/jaeger/internal/storage/cassandra/config"
 	escfg "github.com/jaegertracing/jaeger/internal/storage/elasticsearch/config"
 	"github.com/jaegertracing/jaeger/internal/storage/metricstore/prometheus"
-	"github.com/jaegertracing/jaeger/internal/storage/v1/badger"
-	"github.com/jaegertracing/jaeger/internal/storage/v1/cassandra"
 	es "github.com/jaegertracing/jaeger/internal/storage/v1/elasticsearch"
-	"github.com/jaegertracing/jaeger/internal/storage/v1/memory"
-	"github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse"
-	"github.com/jaegertracing/jaeger/internal/storage/v2/grpc"
 )
 
 var (
 	_ xconfmap.Validator  = (*Config)(nil)
-	_ confmap.Unmarshaler = (*TraceBackend)(nil)
 	_ confmap.Unmarshaler = (*MetricBackend)(nil)
 )
 
-// Config contains configuration(s) for jaeger trace storage.
-// Keys in the map are storage names that can be used to refer to them
-// from other components, e.g. from jaeger_storage_exporter or jaeger_query.
-// We tried to alias this type directly to a map, but conf did not populated it correctly.
-// Note also that the Backend struct has a custom unmarshaler.
+// Config extends the shared storage configuration with metric backends.
 type Config struct {
-	TraceBackends  map[string]TraceBackend  `mapstructure:"backends"`
-	MetricBackends map[string]MetricBackend `mapstructure:"metric_backends"`
-}
-
-// TraceBackend contains configuration for a single trace storage backend.
-type TraceBackend struct {
-	Memory        *memory.Configuration     `mapstructure:"memory"`
-	Badger        *badger.Config            `mapstructure:"badger"`
-	GRPC          *grpc.Config              `mapstructure:"grpc"`
-	Cassandra     *cassandra.Options        `mapstructure:"cassandra"`
-	Elasticsearch *escfg.Configuration      `mapstructure:"elasticsearch"`
-	Opensearch    *escfg.Configuration      `mapstructure:"opensearch"`
-	ClickHouse    *clickhouse.Configuration `mapstructure:"clickhouse"`
+	TraceBackends  map[string]storageconfig.TraceBackend `mapstructure:"backends"`
+	MetricBackends map[string]MetricBackend              `mapstructure:"metric_backends"`
```

Comment:
instead of declaring backend maps explicitly this struct should simply embed Config from cmd/internal/storageconfig/config.go

Reply:
this is not what I meant. `storageconfig.Config already declares the two maps, there is no need to repeat them here, instead just embed that full config:
```
type Config struct {
  storageconfig.Config `mapstructure:",squash"`
}